src/yunohost/domain.py

```diff
@@ -179,9 +179,15 @@ def domain_remove(operation_logger, domain, force=False):
             raise YunohostError('domain_cannot_remove_main_add_new_one', domain=domain)
 
     # Check if apps are installed on the domain
-    app_settings = [_get_app_settings(app) for app in _installed_apps()]
-    if any(s["domain"] == domain for s in app_settings):
-        raise YunohostError('domain_uninstall_app_first')
+    apps_on_that_domain = []
+
+    for app in _installed_apps():
+        settings = _get_app_settings(app)
+        if settings.get("domain") == domain:
+            apps_on_that_domain.append("%s (on https://%s%s)" % (app, domain, settings.get("path")))
```

Comment:
Not all apps are webapps tho :s 

```suggestion
            apps_on_that_domain.append("%s (on https://%s%s)" % (app, domain, settings["path"]) if "path" in settings else app))
```

Reply:
Ah, this is going to do a "https://domain.tld/$app_id" so we actually even want something else :<

Reply:
Oupsi, spoke too fast